guides/example-app-walkthrough.md

```diff
@@ -0,0 +1,217 @@
+
+- `app/getters.js` - This is used to map our routes to the Layout component as props.
+
+
+---
+dogstack expects `routes.js` to be an exported routes configuration in this format:
+
+```js
+
+export default  [
+  {
+    name: 'home',
+    path: '/',
+    exact: true,
+    Component: Home,
+    selector: getIsNotAuthenticated,
+    navigation: {
+      title: 'app.home',
+      icon: 'fa fa-home'
+    }
+  },
+  {
+    name: 'signIn',
+    path: '/sign-in',
+    exact: true,
+    Component: UserIsNotAuthenticated(SignIn),
+    navigation: {
+      title: 'agents.signIn',
+      selector: getIsNotAuthenticated,
+      icon: 'fa fa-sign-in'
+    }
+  }
+]
+```
+
+You can read more into react-router and find the docs [here](https://reacttraining.com/react-router/web/guides/philosophy)
+
+#### store.js - required by dogstack
+
+This is for our [redux]() store config.
+
+- [`updater`(add link to updater.js below)](): a function of shape `action => state => nextState`, combined from each topic using [`redux-fp.concat`](https://github.com/rvikmanis/redux-fp/blob/master/docs/API.md#concat)
+- [`epic`(add link to epic below)](): a function of shape `(action$, store, { feathers }) => nextAction$`, combined from each topic using [`combineEpics`](https://redux-observable.js.org/docs/api/combineEpics.html)
+- [`middlewares`](http://redux.js.org/docs/Glossary.html#middleware): an array of functions of shape `store => next => action`
+- [`enhancers`](http://redux.js.org/docs/Glossary.html#store-enhancer): an array of functions of shape `createStore => createEnhancedStore
+
+```js
+// store.js
+import updater from './updater'
+import epic from './epic'
+
+const middlewares = []
+const enhancers = []
+
+export default {
+  updater,
+  epic,
+  middlewares,
+  enhancers
+}
+```
+
+
+#### updater.js
+
+> An Updater is a type of function that takes an action and returns another function that takes the state and produces a result. Yes, they are just curried reducers with reversed argument order.
+
+If you're used to the term `reducer`, you can see in our `updater.js` file we have a function that converts reducers to updaters
+
+```js
+function reducerToUpdater (reducer) {
+  return action => state => reducer(state, action)
+}
+``` 
+
+
+This file combines all of our apps `updaters` using [`redux-fp.concat`](https://github.com/rvikmanis/redux-fp/blob/master/docs/API.md#concat). Think of this as your 'root updater'. For any updaters to interact with the state, they must be in this file.
+
+#### epic.js
+
+This file exports a function of shape `(action$, store, { feathers }) => nextAction$`, combined from each topic using [`combineEpics`](https://redux-observable.js.org/docs/api/combineEpics.html)
+
+> An Epic is a function which takes a stream of actions and returns a stream of actions. Actions in, actions out.
+
+Need this looked at, still confusing to me after looking into `dogstack-agents`
```

Comment:
epics (via `redux-observable`) are an abstraction in the same space as `redux-thunk` or `redux-saga` or `redux-loop`, it's meant to address the functionality missing in redux for _async_ actions. epics allow us to observe the stream of actions, so we can do async operations like fetching or saving data, by listening for when the initial action is dispatched, then dispatching a series of sync (update) actions based on the progress of the async operation.